build.gradle

```diff
@@ -119,8 +119,17 @@ ext {
   addParametersForTests = { name, options ->
     // -parameters generates arguments with parameter names in TestInfo#getDisplayName.
     // ref: https://github.com/junit-team/junit5/blob/4c0dddad1b96d4a20e92a2cd583954643ac56ac0/junit-jupiter-params/src/main/java/org/junit/jupiter/params/ParameterizedTest.java#L161-L164
-    if (name == "compileTestJava" || name == "compileTestScala")
+    if (name == "compileTestJava" || name == "compileTestScala") {
       options.compilerArgs << "-parameters"
+    } else if (name == "compileScala") {
+      options.compilerArgs << "-Xlint:all"
+      options.compilerArgs << "-encoding" << "UTF-8"
+      options.compilerArgs << "-Xlint:-rawtypes"
+      options.compilerArgs << "-Xlint:-serial"
+      options.compilerArgs << "-Xlint:-try"
+      options.compilerArgs << "-Werror"
+      options.compilerArgs << "-Xlint:-options"
```

Comment:
why we need this option?

Reply:
The `-Xlint:-options` is "Warns about the issues relating to use of command line options" .  [doc](https://docs.oracle.com/en/java/javase/17/docs/specs/man/javac.html)
I think we don't need this option, I will remove it.